### PR TITLE
[dualtor] Add icmp_responder

### PR DESCRIPTION
**Pull request by lolyu**
Add icmp_responder that runs over ptf to reply ICMP echo-request packets
sent by the ToR's LinkProber to simulate a physical server.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Simulate the physical servers for dualtor testbeds that could reply ICMP echo-request(heatbeat).

#### How did you do it?
Add a service `icmp_responder` to `ptf` container.

#### How did you verify/test it?
demo test
```python
import pytest
from tests.common.fixtures.ptfhost_utils import run_icmp_responder


pytestmark = [
    pytest.mark.topology("any"),
]


TARGET_SERVER = "192.168.0.100/21"
DUT_PORT = "Ethernet8"


@pytest.fixture(scope='function', autouse=True)
def setup_intf(ptfhost):
    ptf_port = "eth2"
    ptfhost.shell("ifconfig %s %s" % (ptf_port, TARGET_SERVER))
    yield ptf_port
    ptfhost.shell('ifconfig %s 0.0.0.0' % ptf_port)


@pytest.fixture(scope="function", autouse=True)
def disable_ping(ptfhost):
    ptfhost.shell("echo '1' > /proc/sys/net/ipv4/icmp_echo_ignore_all")
    yield
    ptfhost.shell("echo '0' > /proc/sys/net/ipv4/icmp_echo_ignore_all")


def test_icmp_responder(duthost):
    duthost.shell("ping 192.168.0.100 -c 10")

```

### CODE REVIEW

File: tests/scripts/icmp_responder.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+def respond_to_icmp_request(socket, request):
+    """Respond to ICMP request."""
+    reply = request.copy()
+    reply[ICMP].type = 0
+    # Force re-generation of the checksum
+    reply[ICMP].chksum = None
+    reply[IP].src, reply[IP].dst = request[IP].dst, request[IP].src
+    reply[IP].chksum = None
+    reply[Ether].src, reply[Ether].dst = request[Ether].dst, request[Ether].src
```
Comment (prsunny) [inline tests/scripts/icmp_responder.py:17]: Does this handle icmp packets coming with Vlan tag?

File: tests/scripts/icmp_responder.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from scapy.all import conf, Ether, ICMP, IP
+from scapy.arch import get_if_hwaddr
+from scapy.data import ETH_P_IP
+from select import select
+
+
+def respond_to_icmp_request(socket, request):
+    """Respond to ICMP request."""
+    reply = request.copy()
+    reply[ICMP].type = 0
```
Comment (tahmed-dev) [inline tests/scripts/icmp_responder.py:12]: You may need to subtract `8` from the ICMP checksum.

**Review by tahmed-dev — APPROVED** [review]
LGTM, please address other reviewers' comments.

File: tests/scripts/icmp_responder.py (hunk 1 above)
Comment (lolyu) [reply tests/scripts/icmp_responder.py:17]: If the packet comes with vlan tag, the reply packet will have the same vlan tag, is this expected?

File: tests/scripts/icmp_responder.py (hunk 2 above)
Comment (lolyu) [reply tests/scripts/icmp_responder.py:12]: Sure, but could you please explain why we need to subtract 8?

File: tests/scripts/icmp_responder.py (hunk 1 above)
Comment (prsunny) [reply tests/scripts/icmp_responder.py:17]: yes, thats expected

**Review by theasianpianist — APPROVED** [review]
Confirmed working with heartbeat functionality

File: tests/scripts/icmp_responder.py (hunk 1 above)
Comment (lolyu) [reply tests/scripts/icmp_responder.py:17]: Thanks, mark as resolved.

File: tests/scripts/icmp_responder.py (hunk 2 above)
Comment (lolyu) [reply tests/scripts/icmp_responder.py:12]: Mark as resolved, `scapy` will take care of the recalculation when setting `chksum` to `None`

**Comment by lolyu** [comment]
Thanks!

**Pull request merged**